hatchery/nextflow.go

```diff
@@ -1075,11 +1082,52 @@ func setupSubnet(subnetName string, cidr string, vpcid string, ec2Svc *ec2.EC2)
 		return exsubnet.Subnets[0].SubnetId, nil
 	}
 
+	// Fetch all availability zones
+	// this is being limited to a region by the ec2svc that gets passed in.
+	describeZonesOutput, err := ec2Svc.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe availability zones: %v", err)
+	}
+
+	var selectedZone string
+	for _, zone := range describeZonesOutput.AvailabilityZones {
+		if *zone.State == "available" {
+			input := &ec2.DescribeInstanceTypeOfferingsInput{
+				LocationType: aws.String("availability-zone"),
+				Filters: []*ec2.Filter{
+					{
+						Name:   aws.String("location"),
+						Values: []*string{aws.String(*zone.ZoneName)},
+					},
+					{
+						Name: aws.String("instance-type"),
+						// TODO: Should this be configurable?
+						Values: []*string{aws.String("g4dn.xlarge")},
```

Comment:
we can collect the list of instance types we need from the config, and check that they are all available https://github.com/uc-cdis/cdis-manifest/blob/f7373a8/brh.data-commons.org/manifests/hatchery/hatchery.json#L216

Reply:
Added in this check now